internal/js/modules/k6/browser/browser/page_mapping.go

```diff
@@ -871,3 +875,52 @@ func parseGetByBaseOptions(
 
 	return a, o
 }
+
+// mapPageRoute maps the requested page.route event to the Sobek runtime.
+func mapPageRoute(vu moduleVU, p *common.Page) func(path sobek.Value, handler sobek.Callable) (*sobek.Promise, error) {
+	return func(path sobek.Value, handler sobek.Callable) (*sobek.Promise, error) {
+		ctx := vu.Context()
+		targetID := p.TargetID()
+
+		// Inject JS regex checker for URL regex pattern matching
+		jsRegexChecker, err := injectRegexMatcherScript(ctx, vu, targetID)
+		if err != nil {
+			return nil, err
+		}
+		pathStr := parseStringOrRegex(path, false)
+
+		// Run the event handler in the task queue to
+		// ensure that the handler is executed on the event loop.
+		tq := vu.get(ctx, targetID)
+		routeHandler := func(route *common.Route) error {
+			done := make(chan bool)
+			var rtnErr error
```

Comment:
```suggestion
			done := make(chan struct{})
			var rtnErr error
```

`chan struct{}` is usually way more optimized for signaling such as this.

But to be honest here I probably will make it an `make(chan error, 1)` and use it to return the error instead of having separate variable that is updated, and have a single slot for the one error we might need.

This also will get around potential races where the context is done, but we just got error back

Reply:
As it's a minor update, I'll do it in a following PR :+1:

Reply:
Updated in https://github.com/grafana/k6/pull/4961/commits/f72be0490cc2d8d8948abc98cd46f29b63d92d37